lib/qless/worker/base.rb

```diff
@@ -15,13 +15,16 @@ module Workers
 
     class BaseWorker
       attr_accessor :output, :reserver, :log_level, :interval, :paused,
-                    :options
+                    :options, :sighup_handler
 
       def initialize(reserver, options = {})
         # Our job reserver and options
         @reserver = reserver
         @options = options
 
+        # SIGHUP handler
+        @sighup_handler = options.fetch(:sighup_handler) { lambda { } }
```

Comment:
There is no good way to implement log_stack_trace within Qless. Or, more precisely, log_stack_trace within signal handler. Because of deadlock protection, in trap() context mutexs and monitors are prohibited. As result most of the out-of-the-box loggers (including Ruby library and syslogger) do not work. There are few other options: write a small file (but where? qless does not have configuration...), log to redis (call('log,...) is impossible because we might be out of job context),... So it looks like a simplest solution to outsource handling. The side benefit is that it becomes universal -- somebody might configure completely different action.